enhancements/installer/compact-plus-worker.md

```diff
@@ -0,0 +1,194 @@
+  during the install process, when the number of workers is 1.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+This should not introduce newer risks other than the existing ones in a compact
+cluster deployment.
+
+## Design Details
+
+### Open Questions [optional]
+
+Nothing at the moment
+
+### Test Plan
+
+
+- Add unit tests to ensure the Infrastructure and Scheduler settings are properly set for the different combinations
+
+- Add a periodic job (nightly?) job to run OpenShift tests on a 4-node cluster
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+This enhancement does not have an impact on existing clusters as it's intended
+to affect only day0 operations.
+
+### Version Skew Strategy
+
+Does not apply
+
+### Operational Aspects of API Extensions
+
+
+- Allows for a 4-nodes, highly available, cluster to be deployed as a day0
+  task. It reduces the time required for such deployments to be created.
+
+- Does not impact general deployment functionalities in a negative way.
+
+- Allowing workloads on the master nodes may affect the master's health if the
+  traffic and workloads are too high.
+
+#### Failure Modes
+
+Same as a compact cluster
+
+#### Support Procedures
+
+- The same support procedures used for a compact cluster should be used for a
+  4-nodes cluster. I would appreciate help filling in this section with more
+  details.
+
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+A 4-nodes cluster (a.k.a 3+1) is generally not a recommended architecture. By
+having schedulable masters users will be routing ingress traffic through the
+masters as well as other workloads. Depending on the amount of traffic, and the
+resources available to the cluster, this may result in degraded functionality
+in the masters, which finally affects the general health of the cluster.
+
+With this enhancement, we will be making it easier for users to create such
+4-nodes cluster deployments, which could make users believe it is supported or
+recommended.
+
+## Alternatives
+
+- Don't set masters as schedulable automatically. Instead, cluster
+  administrators will have to explicitly set masters as schedulable in the
+  deployment configuration. The rest of the proposal would remain the same,
+  switching the cluster to `ControlPlaneTopology` if there are 3 schedulable
+  masters and 1 worker.
+
+- Don't do anything and force users to do this in a 2 steps operation: first
+  deploy a compact cluster and then add a worker node.
```

Comment:
Is there no way to do this just by adding/editing manifests used by the installer? So, not in `install-config` but in a separate manifest?

Reply:
you can certainly do this by modifying manifests.

Reply:
After you run `openshift-install create manifests --dir $INST_DIR`, there will be worker machineset manifests under `$INST_DIR/openshift` where you can change the number of worker replicas desired.

Reply:
Or you can specify a single worker replica in the install-config then modify `manifests/cluster-scheduler-02-config.yml` after running `create manifests` IIRC

Reply:
> Is it possible to overwrite the Status using manifests?

Yes, you can set the status in the cluster-infrastructure-02-config.yml manifest.

Reply:
> > Is it possible to overwrite the Status using manifests?
> 
> Yes, you can set the status in the cluster-infrastructure-02-config.yml manifest.

Then I must have done something wrong. I will retry and get back. Thanks for the feedback